redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseAppContent.kt

```diff
@@ -388,50 +394,44 @@ private class ViewContentCodeBinding<A : AppService>(
       val scopedAppService = serviceScope.apply(session.appService)
       val treehouseUi = contentSource.get(scopedAppService)
       treehouseUiOrNull = treehouseUi
-      val restoredId = viewOrNull?.stateSnapshotId
-      val restoredState = if (restoredId != null) codeHost.stateStore.get(restoredId.value.orEmpty()) else null
+      stateSnapshot = viewOrNull?.stateSnapshotId?.let {
+        codeHost.stateStore.get(it.value.orEmpty())
+      }
       try {
-        treehouseUi.start(
-          changesSink = this@ViewContentCodeBinding,
-          onBackPressedDispatcher = onBackPressedDispatcherService,
-          uiConfigurations = uiConfigurationFlow,
-          stateSnapshot = restoredState,
-        )
+        treehouseUi.start(this@ViewContentCodeBinding)
       } catch (e: ZiplineApiMismatchException) {
```

Comment:
We’ve got 3 versions of this function:
  3 args
  4 args
  Host

If Host isn’t found this falls back to the 3-args form, losing support for BackPressed for some cases. I don’t believe that feature is widely used so I view this shortcut as being economical with code.

Reply:
I'm assuming "losing support for BackPressed for some cases" means "losing support for `BackHandler` for some cases."

I'm in the midst of adding native back support to our internal flows project. It sounds frustrating from a user perspective if the custom back override only occurs sometimes.

Reply:
The situation where we lose back handling is when a user’s host app is on a release that has the 4-parameter overload and they guest app has this new interface.

I don’t expect that to impact any of Cash App’s production users. I didn’t see any uses of BackHandler in our production code?

If I’m mistaken it’s easy to restore that functionality! I just need another pair of try/catch clauses to support falling back to that intermediate case.

Reply:
Gotcha, we don't have any usage of `BackHandler` within production code right now, but we will very soon (like within a week or so).

Just to double check, if we:

1. Merge this PR
2. Do an internal release containing these changes
3. Track the new release in all our repos
4. If we then use and ship usages of `BackHandler`, the `BackHandler` will work in ALL cases (i.e., as before)?

I just want to make sure that we don't get into a weird state where the `BackHandler` is invoked inconsistently for users where it should work!

Reply:
Yep, exactly.